crates/http/src/backend/h3/mod.rs

```diff
@@ -0,0 +1,133 @@
+use std::fmt::{Debug, Display};
+use std::net::SocketAddr;
+use std::sync::Arc;
+
+use body::QuicIncomingBody;
+use h3_quinn::BidiStream;
+use scuffle_context::ContextFutExt;
+use utils::copy_response_body;
+
+use crate::error::Error;
+use crate::service::{HttpService, HttpServiceFactory};
+
+pub mod body;
+mod utils;
+
+#[derive(Debug, Clone)]
+pub struct Http3Backend {
+    pub ctx: scuffle_context::Context,
+    pub bind: SocketAddr,
+}
+
+impl Http3Backend {
+    pub async fn run<S>(self, service_factory: S, mut rustls_config: rustls::ServerConfig) -> Result<(), Error<S>>
+    where
+        S: HttpServiceFactory + Clone + Send + 'static,
+        S::Error: Debug + Display,
+        S::Service: Clone + Send + 'static,
+        <S::Service as HttpService>::Error: std::error::Error + Debug + Display + Send + Sync,
+        <S::Service as HttpService>::ResBody: Send,
+        <<S::Service as HttpService>::ResBody as http_body::Body>::Data: Send,
+        <<S::Service as HttpService>::ResBody as http_body::Body>::Error: std::error::Error + Send + Sync,
+    {
+        #[cfg(feature = "tracing")]
+        tracing::debug!("starting server");
+
+        // not quite sure why this is necessary but it is
+        rustls_config.max_early_data_size = u32::MAX;
+        let crypto = h3_quinn::quinn::crypto::rustls::QuicServerConfig::try_from(rustls_config)?;
+        let server_config = h3_quinn::quinn::ServerConfig::with_crypto(Arc::new(crypto));
+
+        let endpoint = h3_quinn::quinn::Endpoint::server(server_config, self.bind)?;
+
+        // handle incoming connections and requests
+        while let Some(new_conn) = endpoint.accept().await {
```

Comment:
Before we had the ability to run multiple connection accepts from different threads, can we add that too? worker count?